tests/mcp/test_tools.py

```diff
@@ -234,3 +252,216 @@ def test_lint_file_with_file_in_result(self, tmp_path: Path):
+
+        # With configuration to allow longer lines
+        result_configured = _lint_file_impl(
+            str(robot_file),
+            select=["LEN08"],
+            configure=["line-too-long.line_length=200"],
+        )
+        has_line_too_long_configured = any(d["rule_id"] == "LEN08" for d in result_configured)
+        assert not has_line_too_long_configured
+
+
+class TestLintContentErrors:
+    """Tests for lint_content error handling."""
+
+    def test_lint_with_invalid_threshold(self):
+        """Test linting with invalid threshold raises error."""
+        content = "*** Test Cases ***\nTest\n    Log    Hello\n"
+        with pytest.raises(ToolError, match="Invalid threshold"):
+            _lint_content_impl(content, threshold="X")
+
+    def test_lint_resource_file(self):
+        """Test linting content as a resource file."""
+        content = "*** Keywords ***\nMy Keyword\n    Log    Hello\n"
+        result = _lint_content_impl(content, filename="test.resource")
+        assert isinstance(result, list)
+
+
+class TestFormatContentErrors:
+    """Tests for format_content error handling."""
+
+    def test_format_with_select(self):
+        """Test formatting with specific formatters selected."""
+        content = "*** Test Cases ***\nTest\n    log  hello\n"
+        result = _format_content_impl(content, select=["NormalizeSeparators"])
+        assert "formatted" in result
+        assert "changed" in result
+
+    def test_format_with_custom_space_count(self):
+        """Test formatting with custom space count."""
+        content = "*** Test Cases ***\nTest\n    Log    Hello\n"
+        result = _format_content_impl(content, space_count=2)
+        assert "formatted" in result
+
+    def test_format_with_custom_line_length(self):
+        """Test formatting with custom line length."""
+        content = "*** Test Cases ***\nTest\n    Log    Hello\n"
+        result = _format_content_impl(content, line_length=80)
+        assert "formatted" in result
+
+
+class TestLintAndFormatIntegration:
+    """Integration tests for lint_and_format functionality."""
+
+    def test_lint_and_format_fixes_issues(self):
+        """Test that formatting can fix some linting issues."""
+        # Content with inconsistent separators
+        content = dedent(
+            """
+            *** Test Cases ***
+            Test
+                log  hello
+                log    world
+        """
+        ).lstrip()
+
+        # Get issues before formatting
+        issues_before = _lint_content_impl(content)
+
+        # Format the content
+        format_result = _format_content_impl(content)
+        assert format_result["changed"] is True
+
+        # Get issues after formatting
+        issues_after = _lint_content_impl(format_result["formatted"])
+
+        # Formatting should have fixed some issues or at least not made it worse
+        assert isinstance(issues_before, list)
+        assert isinstance(issues_after, list)
```

Comment:
The comment suggest we want to check if number of issues did no go up. But it looks we only check for type.

Maybe it's missing?:

```
assert len(issues_after) <= len(issues_before)
```